xla/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -4590,7 +4910,8 @@ absl::StatusOr<CudnnGraph> GetCudnnFlashAttentionOperationGraph(
     const std::optional<dnn::TensorDescriptor> page_table_v_descriptor,
     double scale, const bool use_dropout,
     const std::optional<double> dropout_rate, const dnn::FMHAMaskKind mask_type,
-    const int sliding_window_length, const int max_seg_per_batch) {
+    const int sliding_window_length, const int max_seg_per_batch,
+    const xla::HloComputation* fwd_comp) {
```

Comment:
We're actively trying to reduce the usage of HLO in runtime components like stream_executor to make it easier to produce compatible compiled XLA programs. From what I understand this function is only getting called from the compiler during cuDNN graph construction, right? If not, can we avoid depending on HLO here?

Reply:
I removed the hlo part here by pass in scoreModFunc ptr here instead. But one problem is that I can't do a static factory function since I need to initialize it separately in graph creation to get correct uid order assigned.